website/content/en/preview/upgrading/v1-migration.md

```diff
@@ -0,0 +1,246 @@
+     annotations:
+       compatibility.karpenter.k8s.aws/v1beta1-ubuntu: amiFamily,blockDeviceMappings
+   spec:
+     amiFamily: AL2
+     amiSelectorTerms:
+     - id: ami-foo
+     blockDeviceMappings:
+     - deviceName: 'dev/sda1'
+       rootVolume: true
+       ebs:
+         encrypted: true
+         volumeType: gp3
+         volumeSize: 20Gi
+   ```
+   **NOTE**: If you do not specify `amiSelectorTerms` before upgrading with an Ubuntu AMI, the conversion webhook will fail. If you do this, downgrade to Karpenter `v0.37.1 and add amiSelectorTerms before upgrading again.
+
+* **Migrating userData**: Consider migrating your userData ahead of the v1.0.0 upgrade. See the description of `amiSelectorTerms` below.
+
+### Before upgrading to `1.1.0`
+
+Apply the following changes to your NodePools and EC2NodeClasses, as appropriate, before upgrading them to `v1.1.0` (though okay to make these changes for `1.0.0`)
+
+* **v1beta1 support gone**: In `1.1.0`, v1beta1 is not supported. Migrate all Karpenter yaml files ([NodePools]({{<ref "../concepts/nodepools" >}}), [EC2NodeClasses]({{<ref "../concepts/nodeclasses" >}}), and so on) to v1.
+Also, know that all resources in the cluster also need to be on v1. It's possible (although unlikely) that some resources still may be stored as v1beta1 in ETCD if no writes had been made to them since the v1 upgrade.
+You could use a tool such as [kube-storage-version-migrator](https://github.com/kubernetes-sigs/kube-storage-version-migrator) to handle this.
+
+* **Remove kubelet objects from NodePools**: Check that NodePools no longer contain the kubelet-configuration annotation (`karpenter.sh/v1beta1-kubelet-conversion` annotation).
+Karpenter will crash if NodePool resources contain this annotation.
+
+* **Remove BootstrapMode annotation**: Karpenter will crash if NodePool resources contain the `BootstrapMode` annotation.
+This annotation is no longer being added. If you are migrating an Ubuntu NodeClass to v1, you need to remove the `karpenter.k8s.aws/v1beta1-ubuntu` annotation.
+
+* **KubeletConfiguration**: If you have multiple NodePools pointing to the same EC2NodeClass that have different kubeletConfigurations,
+then you have to manually intervene to add more EC2NodeClasses and point their NodePools to them.
+Otherwise, this will induce drift and you will have to roll your cluster.
+If you have multiple NodePools pointing to the same EC2NodeClass, but they have the same configuration, then you can proceed with the migration
+without having drift or having any additional NodePools or EC2NodeClasses configured.
+
+* **amiSelectorTerms and amiFamily**: For v1, `amiFamily` is no longer required if you instead specify an `alias` in `amiSelectorTerms` in your `EC2NodeClass`. However, you need to update those settings if you are using:
+   * A Custom amiFamily. You must ensure that the node is registered with the `karpenter.sh/unregistered` taint.
+   * An Ubuntu AMI, as described earlier.
+
+### Update metrics
```

Comment:
An update that we made for topologySpreadConstraints to be DoNotSchedule for zone instead of ScheduleAnyways is also not covered here and is something that we should add